morpheus/_lib/src/messages/multi_inference.cpp

```diff
@@ -131,7 +66,7 @@ std::shared_ptr<MultiInferenceMessage> MultiInferenceMessageInterfaceProxy::init
 
 std::shared_ptr<morpheus::InferenceMemory> MultiInferenceMessageInterfaceProxy::memory(MultiInferenceMessage &self)
 {
-    return self.memory;
+    return DCHECK_NOTNULL(std::dynamic_pointer_cast<morpheus::InferenceMemory>(self.memory));
```

Comment:
Can this be a `std::static_pointer_cast` instead? We can add a `DCHECK` beforehand that does the dynamic cast just to be sure.